lib/cylc/network/https/base_client.py

```diff
@@ -96,152 +123,171 @@ def _get_data_from_url(self, url, json_data, method=None):
             if version < [2, 4, 2]:
                 requests_ok = False
         if requests_ok:
-            return self._get_data_from_url_with_requests(
-                url, json_data, method=method)
-        return self._get_data_from_url_with_urllib2(
-            url, json_data, method=method)
+            return self._get_data_from_url_with_requests(http_requests)
+        return self._get_data_from_url_with_urllib2(http_requests)
 
-    def _get_data_from_url_with_requests(self, url, json_data, method=None):
+    def _get_data_from_url_with_requests(self, http_requests):
         import requests
         from requests.packages.urllib3.exceptions import InsecureRequestWarning
         warnings.simplefilter("ignore", InsecureRequestWarning)
         username, password = self._get_auth()
         auth = requests.auth.HTTPDigestAuth(username, password)
         if not hasattr(self, "session"):
             self.session = requests.Session()
-        if method is None:
-            method = self.METHOD
-        if method == self.METHOD_POST:
-            session_method = self.session.post
-        else:
-            session_method = self.session.get
-        try:
-            ret = session_method(
-                url,
-                json=json_data,
-                verify=self._get_verify(),
-                proxies={},
-                headers=self._get_headers(),
-                auth=auth,
-                timeout=self.timeout
-            )
-        except requests.exceptions.SSLError as exc:
-            if "unknown protocol" in str(exc) and url.startswith("https:"):
-                # Server is using http rather than https, for some reason.
-                sys.stderr.write(WARNING_NO_HTTPS_SUPPORT.format(exc))
-                return self._get_data_from_url_with_requests(
-                    url.replace("https:", "http:", 1), json_data)
-            if cylc.flags.debug:
-                import traceback
-                traceback.print_exc()
-            raise ConnectionError(url, exc)
-        except requests.exceptions.Timeout as exc:
-            if cylc.flags.debug:
-                import traceback
-                traceback.print_exc()
-            raise ConnectionTimeout(url, exc)
-        except requests.exceptions.RequestException as exc:
-            if cylc.flags.debug:
-                import traceback
-                traceback.print_exc()
-            raise ConnectionError(url, exc)
-        if ret.status_code == 401:
-            raise ConnectionDeniedError(url, self.prog_name,
-                                        self.ACCESS_DESCRIPTION)
-        if ret.status_code >= 400:
-            from cylc.network.https.util import get_exception_from_html
-            exception_text = get_exception_from_html(ret.text)
-            if exception_text:
-                sys.stderr.write(exception_text)
+
```

Comment:
Yes, it should use the same connection according to the requests documentation: http://docs.python-requests.org/en/master/user/advanced/